src/main/java/org/example/intro/service/impl/CartServiceImpl.java

```diff
@@ -0,0 +1,112 @@
+package org.example.intro.service.impl;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.example.intro.dto.cart.CartItemDto;
+import org.example.intro.dto.cart.CreateCartItemDto;
+import org.example.intro.dto.cart.ShoppingCartDto;
+import org.example.intro.dto.cart.UpdateCartItemsQuantityDto;
+import org.example.intro.mapper.CartItemMapper;
+import org.example.intro.mapper.CartMapper;
+import org.example.intro.model.Book;
+import org.example.intro.model.CartItem;
+import org.example.intro.model.ShoppingCart;
+import org.example.intro.model.User;
+import org.example.intro.repository.book.BookRepository;
+import org.example.intro.repository.cart.CartItemRepository;
+import org.example.intro.repository.cart.CartRepository;
+import org.example.intro.sequrity.CustomUserDetailsService;
+import org.example.intro.service.CartService;
+import org.springframework.data.domain.Page;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+import org.springframework.data.domain.Pageable;
+
+@RequiredArgsConstructor
+@Service
+public class CartServiceImpl implements CartService {
```

Comment:
all logic with shop cart should be in ShoppingCartService only

Reply:
++, renamed